### ISSUE TITLE
Rcpp 0.11.5 test failures with new Rtools toolchain

### ISSUE THREAD

**Issue by kevinushey** [RcppCore/Rcpp#277]
Posting here for posterity; will try to investigate further ASAP...

```
Executing test function test.attributes  ... make: *** No rule to make target `C:\Users\Kevin\R-src\trunk\library\Rcpp\unitTests\cpp\attributes.o', needed by `s
ourceCpp_44614.dll'.  Stop.
Warning message:
running command 'make -f "C:/Users/Kevin/R-src/trunk/etc/x64/Makeconf" -f "C:/Users/Kevin/R-src/trunk/share/make/winshlib.mk" SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)
' SHLIB_LD='$(SHLIB_CXXLD)' SHLIB="sourceCpp_44614.dll" WIN=64 TCLBIN=64 OBJECTS="C:\Users\Kevin\R-src\trunk\library\Rcpp\unitTests\cpp\attributes.o attributes.
o"' had status 2

Error in sourceCpp(file.path(pathToRcppTests, "cpp", file)) :
  Error 1 occurred building shared library.
```

After this failure, R.exe spins here:

```
Executing test function test.DataFrame.AttributeProxy  ...
```

**Comment by kevinushey** [comment]
The first issue here is that `sourceCpp("cpp/attributes.cpp")` fails, for the unit tests, on Windows (for me). The following call to `R CMD SHLIB` is generated for me:

```
C:/Users/Kevin/R-src/trunk/bin/x64/R CMD SHLIB -o "sourceCpp_58407.dll" "C:\Users\Kevin\git\Rcpp\inst\unitTests\cpp\attributes.cpp" "attributes.cpp"
```

@jjallaire, it looks like this is related to the `sourceCpp()` 'find -recursive-dependencies' behaviour added in -- does that sound correct?

In particular, using Rcpp 0.11.4, and running `sourceCpp` on the old `attributes.cpp` file, seems to work fine.

**Comment by jjallaire** [comment]
Yes that sounds right to me. There is someplace where I try to de-dup
translation units but it looks like this isn't kicking in for the main
translation unit

On Mar 15, 2015, at 3:45 PM, Kevin Ushey <notifications@github.com> wrote:

The first issue here is that sourceCpp("cpp/attributes.cpp") fails, for the
unit tests, on Windows (for me). The following call to R CMD SHLIB is
generated for me:

C:/Users/Kevin/R-src/trunk/bin/x64/R CMD SHLIB -o
"sourceCpp_58407.dll"
"C:\Users\Kevin\git\Rcpp\inst\unitTests\cpp\attributes.cpp"
"attributes.cpp"

@jjallaire <https://github.com/jjallaire>, it looks like this is related to
the sourceCpp() 'find -recursive-dependencies' behaviour added in -- does
that sound correct?

In particular, using Rcpp 0.11.4, and running sourceCpp on the old
attributes.cpp file, seems to work fine.

—
Reply to this email directly or view it on GitHub
<https://github.com/RcppCore/Rcpp/issues/277#issuecomment-81213865>.

**Comment by jjallaire** [comment]
I've submitted a fix so that the test now passes (it was passing before on
other platforms but not Windows).

**Comment by coatless** [comment]
This issue was addressed in PR #274.

Please close. (Tagged #506)
